text/0000-try-clone-trait.md

```diff
@@ -0,0 +1,47 @@
+- try-clone-trait
+- Start Date: 2017-10-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+A number of types in the standard library has a `try_clone` method that returns a 
+`Result<T>`. The semantics for conversion is similar in all these cases, thus it
+will be useful to add a new `TryClone` trait and have the types implement that trait.
+
+# Motivation
+A number of crates have defined `TryClone` as a trait and implemented it on standard
+types [1]. Including this in the standard library will reduce some clutter in the code
+of those crates.
```

Comment:
What code? Can you give an example of real-world code that would be improved here?